erpnext/projects/doctype/task/task.py

```diff
@@ -45,6 +46,12 @@ def validate_dates(self):
             frappe.throw(_("{0} can not be greater than {1}").format(frappe.bold("Actual Start Date"), \
                 frappe.bold("Actual End Date")))
 
+	def validate_parent_expected_end_date(self):
+		if self.parent_task:
+			parent_exp_end_date = frappe.db.get_value("Task", self.parent_task, "exp_end_date")
+			if getdate(self.get('exp_end_date')) > getdate(parent_exp_end_date):
+				frappe.throw(_("Expected End Date should be less than or equal to parents task's Expected End Date {0}.").format(getdate(parent_exp_end_date)))
```

Comment:
```suggestion
			if parent_exp_end_date and getdate(self.get("exp_end_date")) > getdate(parent_exp_end_date):
				frappe.throw(_("Expected End Date should be less than or equal to parent task's Expected End Date {0}.").format(getdate(parent_exp_end_date)))
```

1. Check if `parent_exp_end_date` is set since the system should allow setting any date for the child task if the parent's end date is missing.
2. Make the quote-style uniform.
3. Minor change in the message.

Reply:
Done